instill/model/v1alpha/model.proto

```diff
@@ -0,0 +1,372 @@
+message UpdateModelRequest {
+  // A model instance
+  Model model = 1 [ (google.api.field_behavior) = REQUIRED ];
+  // A model version
+  uint64 version = 2 [ (google.api.field_behavior) = REQUIRED ];
+  // Update mask for a model instance
+  google.protobuf.FieldMask update_mask = 3
+      [ (google.api.field_behavior) = REQUIRED ];
+}
+
+// UpdateModelResponse represents a response for a model instance
+message UpdateModelResponse {
+  // A model instance
+  Model model = 1 [ (google.api.field_behavior) = REQUIRED ];
+}
+
+// ListModelRequest represents a request to list models
+message ListModelRequest {}
+
+// ListModelResponse represents a response for a list of models
+message ListModelResponse {
+  // A list of models
+  repeated Model models = 1;
+}
+
+// GetModelRequest represents a request to query a model
+message GetModelRequest {
+  // Model name
+  string name = 1 [ (google.api.field_behavior) = REQUIRED ];
+}
+
+// GetModelResponse represents a response for a model instance
+message GetModelResponse {
+  // A model instance
+  Model model = 1 [ (google.api.field_behavior) = REQUIRED ];
+}
+
+// DeleteModelRequest represents a request to delete a model
+message DeleteModelRequest {
+  // Model name
+  string name = 1 [ (google.api.field_behavior) = REQUIRED ];
+}
+
+// DeleteModelResponse represents an empty response
+message DeleteModelResponse {}
+
+// DeleteModelVersionRequest represents a request to delete a model version
+message DeleteModelVersionRequest {
+  // Model name
+  string name = 1 [ (google.api.field_behavior) = REQUIRED ];
+  // Model version
+  uint64 version = 2 [ (google.api.field_behavior) = REQUIRED ];
+}
+
+// DeleteModelVersionResponse represents an empty response
+message DeleteModelVersionResponse {}
+
+// Input represents the input to a model
+message Input {
+  // Input type
+  oneof type {
+    // Image type URL
+    string image_url = 1;
+    // Image type base64
+    string image_base64 = 2;
+  }
+}
+
+// Output represents the output from a model
+message Output {
+  // Struct output
+  google.protobuf.Struct output = 1;
+}
+
+// TriggerModelRequest represents a request to trigger a model
+message TriggerModelRequest {
+  // Model name
+  string name = 1 [ (google.api.field_behavior) = REQUIRED ];
+  // Model version
+  uint64 version = 2 [ (google.api.field_behavior) = REQUIRED ];
+  // Input to the model
+  repeated Input inputs = 3 [ (google.api.field_behavior) = REQUIRED ];
+}
+
+// TriggerModelResponse represents a response for the output of a model
+message TriggerModelResponse {
+  // Output from a model
+  Output output = 1;
+}
+
+// TriggerModelBinaryFileUploadRequest represents a request to trigger a model
+message TriggerModelBinaryFileUploadRequest {
+  // Model name
+  string name = 1 [ (google.api.field_behavior) = REQUIRED ];
+  // Model version
+  int32 version = 2 [ (google.api.field_behavior) = REQUIRED ];
+  // Model content in byte chunk
```

Comment:
@pingli is this binary infererenced image file or model file? I think it it is binary image file because response message is output

Reply:
Yes, this message is for a multipart-form image request.